### PR TITLE
 Added missing file path parameter in AbstractGenerateTypeService.Editor...

### PR DESCRIPTION
**Pull request by mkrueger**
....

Without it the workspace implementation doesn't know where to save the newly generated document.

### CODE REVIEW

**Comment by mkrueger** [comment]
I'm not sure if it's required for VS.NET. But we in MD need it.

File: src/Features/Core/GenerateType/AbstractGenerateTypeService.Editor.cs [hunk 1]
```diff
@@ -328,7 +328,8 @@ private void AddFoldersToNamespaceContainers(List<string> container, IList<strin
                     documentId,
                     documentName,
                     containers,
-                    sourceCodeKind));
+                    sourceCodeKind,
+                    filePath: Path.Combine (Path.GetDirectoryName (generatingDocument.FilePath), documentName)));
```
Comment (VladimirReshetnikov) [inline src/Features/Core/GenerateType/AbstractGenerateTypeService.Editor.cs]: Please do not use spaces after method names.

Comment (mkrueger) [inline src/Features/Core/GenerateType/AbstractGenerateTypeService.Editor.cs]: That's the years of the mono formatting :/

I'm not sure if it's an issue for VS.NET or not.

**Comment by pharring** [comment]
@dotnet-bot Retest this, please.
@Pilchie Route this, please.

**Comment by Pilchie** [comment]
This is actually the analyzer team, so handing over to @srivatsn

**Comment by srivatsn** [comment]
@jmarolf  @shyamnamboodiripad  can you please take a look?

**Comment by jmarolf** [comment]
The Visual Studio workspace has logic to add the document to a location based on the containers that are passed in, so we never look at the filepath.  In general we don't want to look at the filepaths, we'd rather the workspace decide where things go.  For example, in an ASP webproject, we'd want files to be added to the App_Code folder by default, and only the workspace can know if we are in that situation. If Mono Develop needs this, then we can go ahead and add it for them.  @mkrueger I'm curious about which workspace implementation MD uses, could we perhaps make the fix there?.

**Comment by jmarolf** [comment]
The Visual Studio Workspace looks at the containers that are passed in and determines where to place a document based on that.  In general we don’t want to deal with filepaths, but instead have the workspace be the one to determine where everything should go.  For example, in websites we want all the generated files to go in the App_Code folder and the workspace\project system are the components that can cleanly determine this.  If Mono Develop needs this then we can add it.  My question is: which workspace does MD use, and would it make more sense to modify that workspace instead?

**Comment by mkrueger** [comment]
It's here:

https://github.com/mono/monodevelop/blob/roslyn/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MonoDevelopWorkspace.cs#L602

Any help on improving the workspace is welcome. The ApplyDocumentAdded only gets the DocumentInfo - how can it cleanly determine where thge source goes.

You do not need to add the patch above - I use it as source and I'm ok with maintaining a custom change set. But I would prefer using the logic you guys are using.

**Comment by jmarolf** [comment]
You can look at the handling In the MSBuildWorkspace [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces.Desktop/Workspace/MSBuild/MSBuildWorkspace.cs,1017).  It should be sufficient for your purposes.  The Visual Studio Workspace has more complex behavior that you can look at [here](http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs,429).

**Comment by mkrueger** [comment]
Found & fixed  it.
We had an issue with our workspace implementation. We didn't set the document folders on creation.

Thanks

**Pull request closed without merging**